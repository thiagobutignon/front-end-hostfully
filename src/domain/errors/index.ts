export * from './email-in-use-error'
export * from './invalid-credentials'
export * from './not-found-error'
export * from './unexpected-erros'
