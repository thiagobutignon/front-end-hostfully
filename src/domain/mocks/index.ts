export * from './booking.mock'
export * from './guest.mock'
export * from './location.mock'
export * from './property.mock'
