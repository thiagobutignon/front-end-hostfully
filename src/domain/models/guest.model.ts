export namespace Guest {
  export type Model = {
    numberOfGuests: number
    guests: Guest.Info[]
  }
  export type Info = {
    name: string
    email: string
  }
}
