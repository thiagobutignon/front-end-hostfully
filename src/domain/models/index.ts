export * from './booking.model'
export * from './guest.model'
export * from './location.model'
export * from './property.model'
