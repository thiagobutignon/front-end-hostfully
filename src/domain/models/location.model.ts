export type LocationModel = {
  street: string
  number: string
  city: string
  country: string
}
