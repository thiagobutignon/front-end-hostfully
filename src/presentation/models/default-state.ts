export type DefaultState = {
  isLoading: boolean
  mainError: string
}