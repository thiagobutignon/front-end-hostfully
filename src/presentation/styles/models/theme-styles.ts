export type ThemeStyles = {
  semanticTokens: ThemeStyles.SemanticTokens
  styles: ThemeStyles.Styles
}
// eslint-disable-next-line @typescript-eslint/no-redeclare
export namespace ThemeStyles {
  export type ColorMode = {
    default: string
    _dark: string
  }

  export type Styles = {
    global: {
      body: {
        bgColor: string
        color: string
      }
    }
  }

  export type SemanticTokens = {
    colors: {
      background: ColorMode
      text: ColorMode
      detail: ColorMode
      buttonHud: ColorMode
      successButton: ColorMode
      error: ColorMode
      delete: ColorMode
      reverseBackground: ColorMode
      reverseText: ColorMode
      iconColor: ColorMode
      reverseIconColor: ColorMode
      borderColor: ColorMode
      cardColor: ColorMode
    }
  }
}
