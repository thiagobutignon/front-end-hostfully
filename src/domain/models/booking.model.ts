import { Guest, PropertyModel } from '@/domain/models'

export namespace Booking {
  export type Model = {
    id: string
    totalPrice: number
    numberOfNights: number
    startDate: Date
    endDate: Date
    hostEmail: string
    guests: Guest.Model
    property: PropertyModel
    guestEmail: string
  }
  export type Params = {
    guestEmail: string
    guests: Guest.Model
    startDate: Date
    endDate: Date
    createdAt: Date
    property: PropertyModel
  }
  export type Result = {
    booking?: Booking.Model[]
    error?: string
  }
}
