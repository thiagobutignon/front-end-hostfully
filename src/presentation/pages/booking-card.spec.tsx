import { RenderResult, render } from '@testing-library/react'

import { Booking } from '@/domain/models'
import BookingCardComponent from '@/presentation/pages/booking-card'

describe('BookingCardComponent', () => {
  let sut: RenderResult
  const mockBooking: Booking.Model = {
    id: '1',
    totalPrice: 500,
    numberOfNights: 2,
    startDate: new Date('2024-01-01'),
    endDate: new Date('2024-01-03'),
    hostEmail: 'host@example.com',
    guestEmail: 'guest@example.com',
    guests: {
      numberOfGuests: 2,
      guests: [{ name: 'John Doe', email: 'john@example.com' }]
    },
    property: {
      id: 'prop1',
      name: 'Seaside Escape',
      bedrooms: 1,
      beds: 1,
      cleaningFee: 1,
      description: 'description',
      image: ['image.jpg'],
      location: {
        city: 'city',
        country: 'country',
        number: '123',
        street: 'street'
      },
      maxGuests: 1,
      pricePerNight: 1,
      roomType: 'Apartment',
      serviceFee: 1,
      status: 'Cancelled'
    }
  }

  const onDelete = jest.fn()
  const onUpdate = jest.fn()
  beforeEach(() => {
    sut = render(
      <BookingCardComponent
        booking={mockBooking}
        onDelete={onDelete}
        onUpdate={onUpdate}
      />
    )
  })
  it('renders without crashing', () => {
    const { getByText } = sut
    expect(getByText(`Booking ID: ${mockBooking.id}`)).toBeInTheDocument()
  })

  it('displays correct booking information', () => {
    const { getByText } = sut

    expect(
      getByText(`Property: ${mockBooking.property.name}`)
    ).toBeInTheDocument()
    expect(getByText(`Email: ${mockBooking.guestEmail}`)).toBeInTheDocument()
    expect(getByText('Start Date: Dec 31, 2023')).toBeInTheDocument()
    expect(getByText('End Date: Jan 2, 2024')).toBeInTheDocument()
    expect(
      getByText(`Number of Guests: ${mockBooking.guests.numberOfGuests}`)
    ).toBeInTheDocument()
    expect(
      getByText(`Number of Nights: ${mockBooking.numberOfNights}`)
    ).toBeInTheDocument()
    expect(
      getByText(`Total price: ${mockBooking.totalPrice}`)
    ).toBeInTheDocument()
  })

  test('matches snapshot', () => {
    const { asFragment } = sut

    expect(asFragment()).toMatchSnapshot()
  })
})
