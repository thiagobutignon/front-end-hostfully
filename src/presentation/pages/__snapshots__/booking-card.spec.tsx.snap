// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingCardComponent matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-jlimmz"
  >
    <div
      class="chakra-stack css-m3jj3s"
    >
      <p
        class="chakra-text css-in3yi3"
      >
        Booking ID: 1
      </p>
      <p
        class="chakra-text css-0"
      >
        Property: Seaside Escape
      </p>
      <p
        class="chakra-text css-0"
      >
        Email: guest@example.com
      </p>
      <div
        class="chakra-stack css-1igwmid"
      >
        <p
          class="chakra-text css-0"
        >
          Start Date: Dec 31, 2023
        </p>
        <p
          class="chakra-text css-0"
        >
          End Date: Jan 2, 2024
        </p>
      </div>
      <p
        class="chakra-text css-0"
      >
        Number of Guests: 2
      </p>
      <p
        class="chakra-text css-0"
      >
        Number of Nights: 2
      </p>
      <p
        class="chakra-text css-0"
      >
        Total price: 500
      </p>
    </div>
    <button
      class="chakra-button css-4xx2wk"
      type="button"
    >
      Delete
    </button>
    <button
      class="chakra-button css-4xx2wk"
      type="button"
    >
      Update
    </button>
  </div>
</DocumentFragment>
`;
